sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/DataFeedLiveTests.cs

```diff
@@ -0,0 +1,897 @@
+            Assert.That(createdDataSource.ConnectionString, Is.EqualTo(expectedConnectStr));
+            Assert.That(createdDataSource.Query, Is.EqualTo("query"));
+        }
+
+        private DataFeed GetDataFeedWithMinimumSetup(string name, DataFeedSource dataSource)
+        {
+            var metrics = new List<DataFeedMetric>() { new ("cost") };
+            var ingestionStartTime = DateTimeOffset.Parse("2020-08-01T00:00:00Z");
+
+            var granularity = new DataFeedGranularity(DataFeedGranularityType.Daily);
+            var schema = new DataFeedSchema(metrics);
+            var ingestionSettings = new DataFeedIngestionSettings(ingestionStartTime);
+
+            return new DataFeed(name, dataSource, granularity, schema, ingestionSettings);
+        }
+
+        private DataFeed GetDataFeedWithOptionalMembersSet(string name, DataFeedSource dataSource)
+        {
+            var metrics = new List<DataFeedMetric>()
+            {
+                new ("cost") { MetricDisplayName = "costDisplayName", MetricDescription = "costDescription" },
+                new ("revenue") { MetricDisplayName = "revenueDisplayName", MetricDescription = "revenueDescription" }
+            };
+            var dimensionColumns = new List<DataFeedDimension>()
+            {
+                new ("city"),
+                new ("category") { DimensionDisplayName = "categoryDisplayName" }
+            };
+            var ingestionStartTime = DateTimeOffset.Parse("2020-08-01T00:00:00Z");
+
+            var granularity = new DataFeedGranularity(DataFeedGranularityType.Custom) { CustomGranularityValue = 1360 };
+            var schema = new DataFeedSchema(metrics) { DimensionColumns = dimensionColumns, TimestampColumn = "timestamp" };
+            var ingestionSettings = new DataFeedIngestionSettings(ingestionStartTime)
+            {
+                IngestionStartOffset = TimeSpan.FromMinutes(30),
+                IngestionRetryDelay = TimeSpan.FromSeconds(80),
+                StopRetryAfter = TimeSpan.FromMinutes(10),
+                DataSourceRequestConcurrency = 5
+            };
+
+            return new DataFeed(name, dataSource, granularity, schema, ingestionSettings)
+            {
+                Description = "This data feed was created to test the .NET client.",
+                AccessMode = DataFeedAccessMode.Public,
+                ActionLinkTemplate = "https://fakeurl.com/%metric/%datafeed",
+                Administrators = new List<string>() { "fake@admin.com" },
+                Viewers = new List<string>() { "fake@viewer.com" },
+                MissingDataPointFillSettings = new () { FillType = DataFeedMissingDataPointFillType.CustomValue, CustomFillValue = 45.0 }
+            };
+        }
+
+        private void ValidateDataFeedWithMinimumSetup(DataFeed dataFeed, string expectedId, string expectedName)
+        {
+            var ingestionStartTime = DateTimeOffset.Parse("2020-08-01T00:00:00Z");
```

Comment:
should this be a private static variable of the class so that it is the same in the test setup and the validation?

Reply:
I guess the same will be true with all other properties, so good for now. 
If it ever becomes a problem, then we will address

Reply:
Yeah, I had it as a class member before, but I realized it would not be worth it doing the same for every property here (and making it way worse once we have the `Update` scenarios).